python_src/src/lamp_py/ingestion/convert_gtfs_rt.py

```diff
@@ -311,28 +284,44 @@ def gz_to_pyarrow(
                 ) as file:
                     json_data = json.load(file)
 
-            # Create empty 'table' as dict of lists for export schema
-            table = self.detail.empty_table()
-
             # parse timestamp info out of the header
             feed_timestamp = json_data["header"]["timestamp"]
             timestamp = datetime.fromtimestamp(feed_timestamp, timezone.utc)
 
-            # for each entity in the list, create a record, add it to the table
-            for entity in json_data["entity"]:
-                record = self.record_from_entity(entity=entity)
-                record.update(
-                    {
-                        "year": timestamp.year,
-                        "month": timestamp.month,
-                        "day": timestamp.day,
-                        "hour": timestamp.hour,
-                        "feed_timestamp": feed_timestamp,
-                    }
-                )
+            table = pyarrow.Table.from_pylist(
+                json_data["entity"], schema=self.detail.import_schema
+            )
 
-                for key, value in record.items():
-                    table[key].append(value)
+            table = table.append_column(
+                "year",
+                pyarrow.array(
+                    [timestamp.year] * table.num_rows, pyarrow.uint16()
+                ),
+            )
+            table = table.append_column(
+                "month",
+                pyarrow.array(
+                    [timestamp.month] * table.num_rows, pyarrow.uint8()
+                ),
+            )
+            table = table.append_column(
+                "day",
+                pyarrow.array(
+                    [timestamp.day] * table.num_rows, pyarrow.uint8()
+                ),
+            )
+            table = table.append_column(
+                "hour",
+                pyarrow.array(
+                    [timestamp.hour] * table.num_rows, pyarrow.uint8()
+                ),
+            )
+            table = table.append_column(
+                "feed_timestamp",
+                pyarrow.array(
+                    [feed_timestamp] * table.num_rows, pyarrow.uint64()
+                ),
+            )
```

Comment:
appends each of our calendar fields to the pyarrow table, based on `feed_timestamp`